examples/text/Text.cpp

```diff
@@ -0,0 +1,840 @@
+    }
+
+    void setStyle(std::uint32_t style)
+    {
+        text.setStyle(style);
+        update();
+    }
+
+    void setTextOrientation(sf::Text::TextOrientation orientation)
+    {
+        text.setTextOrientation(orientation);
+        update();
+    }
+
+    void colorTextDirection(bool enable)
+    {
+        colorDirection = enable;
+    }
+
+    void drawBoundingBox(sf::RenderTarget& target) const
+    {
+        target.draw(boundingBox);
+    }
+
+    void drawGlyphBox(sf::RenderTarget& target) const
+    {
+        target.draw(glyphBox, text.getTransform());
+    }
+
+    void drawCursor(sf::RenderTarget& target) const
+    {
+        target.draw(cursor, text.getTransform());
+    }
+
+    void drawText(sf::RenderTarget& target) const
+    {
+        target.draw(text);
+    }
+
+    void update()
+    {
+        // Update text bounding box
+        const auto localBounds = text.getLocalBounds();
+        boundingBox.setPosition(text.getPosition() + localBounds.position);
+        boundingBox.setSize(localBounds.size);
+
+        // Position the glyph bounding box and cursor
+        updateGlyphBox(0);
+        computeCursorPositions();
+        updateCursorPosition(0);
+
+        // Update rainbow text default y positions
+        const auto& vertexData = text.getVertexData();
+        rainbowYPositions.resize(vertexData.getVertexCount());
+        std::transform(vertexData.begin(),
+                       vertexData.end(),
+                       rainbowYPositions.begin(),
+                       [](const sf::Vertex& vertex) { return vertex.position.y; });
+    }
+
+    void updateGlyphBox(int offset)
+    {
+        const auto& shapedGlyphs = text.getShapedGlyphs();
+
+        glyphIndex = static_cast<std::size_t>(
+            std::clamp(static_cast<int>(glyphIndex) + offset, 0, static_cast<int>(shapedGlyphs.size()) - 1));
+
+        const auto& shapedGlyph = shapedGlyphs[glyphIndex];
+        const auto& localBounds = text.getLocalBounds();
+        const auto  boxHeight   = static_cast<float>(text.getCharacterSize());
+
+        // Check if we are dealing with whitespace or a glyph with a texture
+        // The bounds of a whitespace glyph will either have no width or no height depending on orientation
+        if (shapedGlyph.glyph.bounds.size.y == 0.0f)
+        {
+            // Horizontal space
+            glyphBox.setPosition({(shapedGlyph.position + shapedGlyph.glyph.bounds.position).x,
+                                  localBounds.position.y + shapedGlyph.position.y - boxHeight});
+            glyphBox.setSize({shapedGlyph.glyph.bounds.size.x, boxHeight});
+            glyphBox.setOutlineColor(sf::Color(0, 127, 255));
+        }
+        else if (shapedGlyph.glyph.bounds.size.x == 0.0f)
+        {
+            // Vertical space
+            glyphBox.setPosition({-localBounds.size.x / 2.0f,
+                                  localBounds.position.y + shapedGlyph.position.y - shapedGlyph.glyph.bounds.size.y});
```

Comment:
```suggestion
                                  localBounds.position.y + shapedGlyph.position.y - boxHeight});
```

Using the boxHeight produces a better result. 
Should we change that?

![Image](https://github.com/user-attachments/assets/8e56caa4-8dcb-41a9-a1c0-027f7a7e488a)